Tensile/Source/client/source/HardwareMonitor.cpp

```diff
@@ -386,9 +394,184 @@ namespace Tensile
+                                                              double(0))) / m_tempHotspotValues.size();
+            }
+            
+            throw std::runtime_error("Can't read temperature value that wasn't requested: ");
+        }
+        
+        void HardwareMonitor::logMinMaxMedianAverage()
+        {
+            assertNotActive();
+            if(m_dataPoints == 0)
+                throw std::runtime_error("No data points collected!");
+           
+            if(m_freqValues.size() == 0 || m_powerValues.size() == 0 || m_tempHotspotValues.size() == 0)
+            {
+                throw std::runtime_error("Frequency, Power, Temperature array is empty.\n"); 
+            }
+            
+            std::cout <<"\nROCm SMI API consolidated frequency,power,temperature data" << "\n" ;   
+            std::cout << "GFX Value\t\t\t\t"<< "PPT0_value\t" << "Temperature\n";
+            // Log individual data  
+            for (int i = 0; i < m_freqValues.size(); i++) 
+            {
+                std::cout<<std::left<<std::setw(13) << m_freqValues[i] << "\t\t\t\t" << m_powerValues[i] << "\t\t\t\t\t" << m_tempHotspotValues[i]<<"\n";
+            }
+            
+            std::cout <<"\n";
+            std::cout <<"\t\t\t\t\t\t\t\tMin\t\t\t"<<"Max\t\t\t"<<"Average\t\t"<<"Median\n";
+           // min, max,avg, median frequency
+            std::cout <<std::left << std::setw(12) << "GFX Frequency\t\t"<<std::setw(8)<< *std::min_element(m_freqValues.begin(), m_freqValues.end());
+            std::cout <<std::setw(8)<< *std::max_element(m_freqValues.begin(), m_freqValues.end());
+            std::cout <<std::setw(10)<< std::accumulate(m_freqValues.begin(), m_freqValues.end(), double(0)) / m_freqValues.size();
+            std::cout << std::setw(8)<< getMedianGfxFrequency() <<"\n";
+
+           //  min, max, avg, median Power
+            std::cout <<std::left<< std::setw(15)<< "Power Value\t\t"<< std::setw(8) << *std::min_element(m_powerValues.begin(), m_powerValues.end());
+            std::cout << std::setw(8)<< *std::max_element(m_powerValues.begin(), m_powerValues.end());
+            std::cout <<std::setw(10)<< std::accumulate(m_powerValues.begin(), m_powerValues.end(), double(0)) / m_powerValues.size();
+            std::cout << std::setw(8)<< getMedianPower()<<"\n";
+            
+           // min, max, avg, median Temperature 
+            std::cout <<std::left<<std::setw(15)<<"Temperature\t\t"<< std::setw(8);
+            std::cout << *std::min_element(m_tempHotspotValues.begin(), m_tempHotspotValues.end());
+            std::cout << std::setw(8)<< *std::max_element(m_tempHotspotValues.begin(), m_tempHotspotValues.end());
+            std::cout << std::setw(10)<< std::accumulate(m_tempHotspotValues.begin(), m_tempHotspotValues.end(), double(0)) / m_tempHotspotValues.size();
+            std::cout << std::setw(10)<< getMedianHotSpotTemperature()<<"\n\n";
```

Comment:
Is it intentional to use setw(10) here? Other locations use setw(8) for median value.

Reply:
updated setw common for avg and median